libraries/AP_Proximity/AP_Proximity_RPLidarA2.cpp

```diff
@@ -97,271 +103,302 @@ void AP_Proximity_RPLidarA2::update(void)
 // get maximum distance (in meters) of sensor
 float AP_Proximity_RPLidarA2::distance_max() const
 {
-    return 16.0f;  //16m max range RPLIDAR2, if you want to support the 8m version this is the only line to change
+    switch (model) {
+    case Model::UNKNOWN:
+        return 0.0f;
+    case Model::A1:
+        return 8.0f;
+    case Model::A2:
+        return 16.0f;
```

Comment:
```suggestion
        return 16.0f; //16m max range RPLIDAR2, if you want to support the 8m version this is the only line to change
```

Reply:
Can this be read from the device?

Reply:
AFAIK, no.

Reply:
PRX1_MIN and MAX exist these days though so it should be possible for users to set this manually.  Another alternative is that we create multiple PRX1_TYPE enum values for the various sensor but they all end up using the same driver.

Reply:
I sugest keeping the 16m to support most A2 devices, but use

min_range = max(PRX1_MIN, 0.05f)
 max_range = min(PRX1_MAX, 16.0f)

Reply:
That's essentially what the current code does.